asv_bench/benchmarks/pandas/frame_methods.py

```diff
@@ -0,0 +1,286 @@
+
+    def time_dict_rename_both_axes(self):
+        self.df.rename(index=self.dict_idx, columns=self.dict_idx)
+
+
+class Repr:
+    def setup(self):
+        nrows = 10000
+        data = np.random.randn(nrows, 10)
+        arrays = np.tile(np.random.randn(3, nrows // 100), 100)
+        idx = MultiIndex.from_arrays(arrays)
+        self.df3 = DataFrame(data, index=idx)
+        self.df4 = DataFrame(data, index=np.random.randn(nrows))
+        self.df_tall = DataFrame(np.random.randn(nrows, 10))
+        self.df_wide = DataFrame(np.random.randn(10, nrows))
+
+    def time_repr_tall(self):
+        repr(self.df_tall)
+
+    def time_frame_repr_wide(self):
+        repr(self.df_wide)
+
+
+class MaskBool:
+    def setup(self):
+        data = np.random.randn(1000, 500)
+        df = DataFrame(data)
+        df = df.where(df > 0)
+        self.bools = df > 0
+        self.mask = isnull(df)
+
+    def time_frame_mask_bools(self):
+        self.bools.mask(self.mask)
+
+    def time_frame_mask_floats(self):
+        self.bools.astype(float).mask(self.mask)
+
+
+class Isnull:
+    def setup(self):
+        N = 10**3
+        self.df_no_null = DataFrame(np.random.randn(N, N))
+
+        sample = np.array([np.nan, 1.0])
+        data = np.random.choice(sample, (N, N))
+        self.df = DataFrame(data)
+
+        sample = np.array(list(string.ascii_letters + string.whitespace))
+        data = np.random.choice(sample, (N, N))
+        self.df_strings = DataFrame(data)
+
+        sample = np.array(
+            [
+                NaT,
+                np.nan,
+                None,
+                np.datetime64("NaT"),
+                np.timedelta64("NaT"),
+                0,
+                1,
+                2.0,
+                "",
+                "abcd",
+            ]
+        )
+        data = np.random.choice(sample, (N, N))
+        self.df_obj = DataFrame(data)
+
+    def time_isnull_floats_no_null(self):
+        isnull(self.df_no_null)
+
+    def time_isnull(self):
+        isnull(self.df)
+
+    def time_isnull_strngs(self):
+        isnull(self.df_strings)
+
+    def time_isnull_obj(self):
+        isnull(self.df_obj)
+
+
+class Dropna:
+
+    params = (["all", "any"], [0, 1])
+    param_names = ["how", "axis"]
+
+    def setup(self, how, axis):
+        self.df = DataFrame(np.random.randn(10000, 1000))
+        self.df.iloc[50:1000, 20:50] = np.nan
+        self.df.iloc[2000:3000] = np.nan
+        self.df.iloc[:, 60:70] = np.nan
+        self.df_mixed = self.df.copy()
+        self.df_mixed["foo"] = "bar"
+
+    def time_dropna(self, how, axis):
+        self.df.dropna(how=how, axis=axis)
+
+    def time_dropna_axis_mixed_dtypes(self, how, axis):
+        self.df_mixed.dropna(how=how, axis=axis)
+
+
+class Dtypes:
+    def setup(self):
+        self.df = DataFrame(np.random.randn(1000, 1000))
+
+    def time_frame_dtypes(self):
+        self.df.dtypes
+
+
+class Equals:
```

Comment:
@anmyachev @YarShev do you think this operation worth to leave as it doesn't do smth that we already benchmark?

Reply:
I do not see we benchmark this so I would suggest leaving it.

Reply:
All cases??

Reply:
We can leave 1 or 2 cases.